src/librbd/ImageCtx.cc

```diff
@@ -795,6 +795,14 @@ class SafeTimerSingleton : public SafeTimer {
       alloc_hint_flags |= librados::ALLOC_HINT_FLAG_INCOMPRESSIBLE;
     }
 
+    op_flags = 0;
+    auto read_policy = config.get_val<std::string>("rbd_read_from_replica_policy");
```

Comment:
You need to verify that the min OSD release Octopus (see `get_min_compatible_osd`)